cypress/e2e/tests/pages/manager/machine-deployments.spec.ts

```diff
@@ -0,0 +1,157 @@
+import PromptRemove from '@/cypress/e2e/po/prompts/promptRemove.po';
+import MachineDeploymentsPagePo from '@/cypress/e2e/po/pages/cluster-manager/machine-deployments.po';
+import * as path from 'path';
+import * as jsyaml from 'js-yaml';
+
+describe('MachineDeployments', { testIsolation: 'off', tags: ['@manager', '@adminUser'] }, () => {
+  const machineDeploymentsPage = new MachineDeploymentsPagePo('_');
+  const runTimestamp = +new Date();
```

Comment:
i think now is the time to resolve the issue of a common approach to making names for resources created by e2e tests.

We need a common function that will create it, suggest something like
```
const runTimestamp = +new Date();
function createE2EResourceName(context: string) {
   return `e2e-test-` + runTimestamp + `-` + context
}
```

We can start using it here and then gradually move the other usages over

Reply:
Good call! I've created a custom command called `createE2EResourceName`. We can use it in the tests via [aliasing](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Sharing-Context). So far I've update `machine-deployments.spec` and `repositories.spec`. I will wait for your feedback before updating the rest of the tests in this PR. Please see updated code below:

custom command: https://github.com/rancher/dashboard/blob/ca300951b879e5acb27ed8b908cd238225ae8129/cypress/support/commands/commands.ts#L66

test usage: https://github.com/rancher/dashboard/blob/ca300951b879e5acb27ed8b908cd238225ae8129/cypress/e2e/tests/pages/manager/machine-deployments.spec.ts#L18

Reply:
Looks good, one minor comment inline. I'll keep this thread open to cover the other places it needs to be called.

And TIL `.as` will assign the response to `this`.